src/jon_mikkel_ex/ex05/bounded_sim.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+
+__author__ = "Jon-Mikkel Korsvik"
+__email__ = "jonkors@nmbu.no"
+
+from src.jon_mikkel_ex.ex05.walker_sim import Walker, Simulation
+
+
+class BoundedWalker(Walker):
+    def __init__(self, start_place, home, left_limit, right_limit):
+        """
+        Initialise the walker
+
+        Arguments
+        ---------
+        start_place : int
+            The walker's initial position
+        home : int
+            The walk ends when the walker reaches home
+        left_limit : int
+            The left boundary of walker movement
+        right_limit : int
+            The right boundary  of walker movement
+        """
+        self.left_limit = left_limit
+        self.right_limit = right_limit
+        super().__init__(start_place, home)
+
+    def is_within_bounds(self):
+        """
+        Checks whether the walkers position is in between the outer limits
+        :return: Boolean
+        """
+        return self.left_limit <= self.get_position() <= self.right_limit
+
+    def move(self):
+        position_before_moving = self.get_position()
+        Walker.move(self)
+        if self.is_within_bounds():
```

Comment:
I would recommend not using a blank `return`here and checking instead `if not self.is_within_bounds(): decrease steps etc....`